test-utils/src/dev/db.rs

```diff
@@ -590,10 +607,34 @@ pub async fn check_db_version() -> Result<(), CockroachStartError> {
 ///
 /// It seems we can infer that "the exit status is not available at this time"
 /// means that the process has not exited.  After all, if it _had_ exited, we'd
-/// fall into the first case.  It's not clear under what conditions this function
-/// could ever fail.  It's not clear from the source that it's even possible.
-fn process_exited(child_process: &mut tokio::process::Child) -> bool {
-    child_process.try_wait().unwrap().is_some()
+/// fall into the first case.  It's not clear under what conditions this
+/// function could ever fail.  It's not clear from the source that it's even
+/// possible.
```

Comment:
It looks like [tokio ultimately calls try_wait from std](https://github.com/tokio-rs/tokio/blob/08d49532b4179d9209a92a5a4c425e63d76218fe/tokio/src/process/unix/mod.rs#L57), which [calls waitpid with WNOHANG](https://github.com/rust-lang/rust/blob/ec55c61305eaf385fc1b93ac9a78284b4d887fe5/library/std/src/sys/unix/process/process_unix.rs#L622).  Per the manpage of waitpid (at least on illumos), it doesn't seem possible for any of the error cases to trigger (absent some weird bug where the process doesn't have the right pid or something like that), at least assuming we can't get `EINTR` if we passed `WNOHANG` (which I'm not sure about?).

Reply:
Yeah, I don't remember the details (this comment didn't actually change here, it's pretty old) but I came to a similar conclusion and left this vague note.